### PR TITLE
feat: Add update payment pipleine to mark payment ready for processing

### PR DESCRIPTION
**Pull request by wdrussell2015**
[THES-204](https://2u-internal.atlassian.net/browse/THES-204)

This PR adds the Update Payment api call to the mark payment as ready for processing pipeline flow.

### CODE REVIEW

File: commerce_coordinator/apps/titan/pipeline.py [hunk 1]
```diff
@@ -218,3 +218,30 @@ def run_filter(self, order_uuid, **kwargs):  # pylint: disable=arguments-differ
...
+
+        api_client = TitanAPIClient()
+        try:
+            response = api_client.update_payment(
+                payment_number=payment_number,
+                payment_state=payment_state,
+                response_code=response_code)
+
+        except HTTPError as exc:
+            raise APIException("Error updating the payment details in titan") from exc
```
Comment (pshiu) [inline commerce_coordinator/apps/titan/pipeline.py:243]: Is there enough information in the exception? Will there a way to connect the exception message with the parameters used to run the PipelineStep? I noticed many other PipelineSteps in the code have a `logger.exception` with the parameters of PipelineStep. Does it make sense to reproduce here?

File: commerce_coordinator/apps/frontend_app_payment/filters.py [hunk 2]
```diff
@@ -118,5 +118,14 @@ def run_filter(cls, **kwargs):
...
-        pipeline_output = super().run_pipeline(**kwargs)
+        pipeline_output = super().run_pipeline(
+            **kwargs,
+        )
+        if 'payment_data' in pipeline_output:
+            payment_data = pipeline_output['payment_data']
+            payment_state_processing_cache_key = get_payment_state_cache_key(
+                payment_data['payment_number'],
+                CachePaymentStates.PROCESSING.value
+                )
```
[the remark this answers is not in the document]
Comment (wdrussell2015) [reply commerce_coordinator/apps/frontend_app_payment/filters.py]: fixed

File: commerce_coordinator/apps/titan/pipeline.py [hunk 3]
```diff
@@ -218,3 +218,30 @@ def run_filter(self, order_uuid, **kwargs):  # pylint: disable=arguments-differ
         update_billing_address_output = BillingAddressSerializer(data=response)
         update_billing_address_output.is_valid(raise_exception=True)
         return {'billing_address_data': update_billing_address_output.data}
+
+
+class UpdateTitanPayment(PipelineStep):
```
[the remark this answers is not in the document]
Comment (wdrussell2015) [reply commerce_coordinator/apps/titan/pipeline.py:223]: [updated](https://github.com/edx/commerce-coordinator-private/pull/60)

File: commerce_coordinator/apps/frontend_app_payment/filters.py [hunk 4]
```diff
@@ -118,5 +118,14 @@ def run_filter(cls, **kwargs):
         Arguments:
             kwargs: arguments passed through from the payment process.
         """
-        pipeline_output = super().run_pipeline(**kwargs)
+        pipeline_output = super().run_pipeline(
+            **kwargs,
+        )
+        if 'payment_data' in pipeline_output:
```
[the remark this answers is not in the document]
Comment (wdrussell2015) [reply commerce_coordinator/apps/frontend_app_payment/filters.py:124]: Added an else to with a log statement that payment data did not exist and could not be cached. Open to suggestions on if we should do anything else.  an error would've thrown in either client or the pipeline step from the serializer for us to not have payment_data in the output

File: commerce_coordinator/apps/titan/pipeline.py (hunk 1 above)
Comment (wdrussell2015) [reply commerce_coordinator/apps/titan/pipeline.py:247]: added

**Pull request merged**